Pointer Lock API tests should run on iOS

aprotyas opened the pull request:
#### 5b90f5e9fa6b0c378370ddacf082d633207d8619
<pre>
Pointer Lock API tests should run on iOS
<a href="https://bugs.webkit.org/show_bug.cgi?id=296958">https://bugs.webkit.org/show_bug.cgi?id=296958</a>
<a href="https://rdar.apple.com/156781157">rdar://156781157</a>

Reviewed by Wenson Hsieh.

This patch makes the pointer lock API tests cross platform. We make a
couple of changes to facilitate this:

- Create a PointerLockTests fixture class. This class does all the
  common test set up work, including instantiating a UIApp, setting up
  the mouse device observer, and manually enabling the Pointer Lock API
  feature for iOS.
- Abstract the notion of a `click` in this test suite, using the
  MouseEventTestHarness to drive mouse interactions on iOS.

We also leave a couple of FIXMEs for future test coverage opportunities.

* Source/WTF/wtf/PlatformEnableCocoa.h:
Drive-by fix: After 297891@main, we were no longer explicitly disabling
ENABLE_POINTER_LOCK on watchOS and tvOS. This meant we were building
the Pointer Lock API on said platforms which is an unwanted behavior
change.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
(-[PointerLockDelegate resetState]):
(-[PointerLockDelegate waitForPointerLockEngaged]):
(-[PointerLockDelegate waitForPointerLockLost]):
(-[PointerLockDelegate _webViewDidRequestPointerLock:completionHandler:]):
(-[PointerLockDelegate _webViewDidLosePointerLock:]):
((PointerLockTests, Simple)):
((PointerLockTests, ClientDisplaysAlertSheetWhilePointerLockActive)):
((WebKit, PointerLock)): Deleted.
((WebKit, ClientDisplaysAlertSheetWhilePointerLockActive)): Deleted.

Canonical link: <a href="https://commits.webkit.org/298276@main">https://commits.webkit.org/298276@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/98b543f5dd371920f361ebdb20c892c02d71cffa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/114942 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/34662 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/25140 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/121066 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/65610 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/b0834ab0-64b5-466d-af03-edcc01f37e7f) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/116831 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/35304 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/43222 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/87337 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/65610 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/92699e7b-e2c4-41dd-8b79-0fc7c00d5b52/bf08746c-4d67-4c5e-8eb3-d46b1f7bd5e1) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/117890 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/28129 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/103196 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/67732 "Passed tests") | | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/27301 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/21315 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/64720 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/107236 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/97510 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/21433 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/124256 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/113488 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/41919 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/31328 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/96138 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/42290 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/99387 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/95922 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/41121 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/18962 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/37956 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/18398 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/41793 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/47326 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/137695 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/41343 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/36807 "Found 18 new JSC stress test failures: stress/arrowfunction-constructor.js.bytecode-cache, stress/arrowfunction-constructor.js.default, stress/arrowfunction-constructor.js.dfg-eager, stress/arrowfunction-constructor.js.dfg-eager-no-cjit-validate, stress/arrowfunction-constructor.js.eager-jettison-no-cjit, stress/arrowfunction-constructor.js.mini-mode, stress/arrowfunction-constructor.js.no-cjit-collect-continuously, stress/arrowfunction-constructor.js.no-cjit-validate-phases, stress/arrowfunction-constructor.js.no-llint, stress/builtin-function-is-construct-type-none.js.bytecode-cache ... (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/44659 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/43086 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
```diff
@@ -1508,3 +1430,169 @@ - (void)webView:(WKWebView *)webView runJavaScriptAlertPanelWithMessage:(NSStrin
+        [m_webView synchronouslyLoadHTMLString:
+            @"<canvas width='800' height='600'></canvas><script>"
+            @"var canvas = document.querySelector('canvas');"
+            @"canvas.onclick = ()=>{canvas.requestPointerLock()};"
```
Small nit — spacing seems a bit inconsistent here.
```suggestion
            @"canvas.onclick = () => canvas.requestPointerLock();"
```

aprotyas replied:
True, was not paying enough attention with the old code. I'll fix it!

aprotyas committed:
Pointer Lock API tests should run on iOS
https://bugs.webkit.org/show_bug.cgi?id=296958
rdar://156781157

Reviewed by Wenson Hsieh.

This patch makes the pointer lock API tests cross platform. We make a
couple of changes to facilitate this:

- Create a PointerLockTests fixture class. This class does all the
  common test set up work, including instantiating a UIApp, setting up
  the mouse device observer, and manually enabling the Pointer Lock API
  feature for iOS.
- Abstract the notion of a `click` in this test suite, using the
  MouseEventTestHarness to drive mouse interactions on iOS.

We also leave a couple of FIXMEs for future test coverage opportunities.

* Source/WTF/wtf/PlatformEnableCocoa.h:
Drive-by fix: After 297891@main, we were no longer explicitly disabling
ENABLE_POINTER_LOCK on watchOS and tvOS. This meant we were building
the Pointer Lock API on said platforms which is an unwanted behavior
change.

* Tools/TestWebKitAPI/Tests/WebKitCocoa/UIDelegate.mm:
(-[PointerLockDelegate resetState]):
(-[PointerLockDelegate waitForPointerLockEngaged]):
(-[PointerLockDelegate waitForPointerLockLost]):
(-[PointerLockDelegate _webViewDidRequestPointerLock:completionHandler:]):
(-[PointerLockDelegate _webViewDidLosePointerLock:]):
((PointerLockTests, Simple)):
((PointerLockTests, ClientDisplaysAlertSheetWhilePointerLockActive)):
((WebKit, PointerLock)): Deleted.
((WebKit, ClientDisplaysAlertSheetWhilePointerLockActive)): Deleted.

Canonical link: https://commits.webkit.org/298276@main
